bin/codemods/remove-linked-state-mixin.js

```diff
@@ -0,0 +1,182 @@
+		return linkPropName === 'valueLink'
+			? { valuePropName: 'value', changeHandlerName: 'handleInputValueChange' }
+			: { valuePropName: 'checked', changeHandlerName: 'handleInputCheckedChange' };
+	}
+
+	// Transform calls to `linkState` into `value` and `onChange` props
+	function transformLinkStateCalls( classPath ) {
+		const changeMethodsToAdd = new Set;
+
+		// Find all usages of 'this.linkState'
+		j( classPath ).find( j.MemberExpression, {
+			object: { type: 'ThisExpression' },
+			property: { type: 'Identifier', name: 'linkState' }
+		} ).forEach( ( linkStateCall ) => {
+			// Verify they are all function calls with one string arg: this.linkState( 'x' )
+			// We can't transform any other occurences.
+			const caller = linkStateCall.parentPath.value;
+			if ( ! j.match( caller, { type: 'CallExpression', arguments: [ { type: 'Literal' } ] } ) ) {
+				console.log( 'ERR: Not a method call with one arg!' );
+				return;
+			}
+
+			// Save the name of the state property that's the target of the link
+			const stateProp = caller.arguments[ 0 ].value;
+
+			// Is the 'this.linkState' call inside a JSX attribute?
+			// We can't transform it if it isn't.
+			const linkProps = j( linkStateCall ).closest( j.JSXAttribute );
+			if ( linkProps.size() === 0 ) {
+				console.log( 'ERR: Not a child of JSX attribute!' );
+				return;
+			}
+
+			const linkProp = linkProps.get();
+			const linkPropName = linkProp.value.name.name;
+			if ( ! [ 'valueLink', 'checkedLink' ].includes( linkPropName ) ) {
+				console.log( 'ERR: The JSX attribute is neither valueLink nor checkedLink!' );
+				return;
+			}
+
+			const openingEl = j( linkProp ).closest( j.JSXOpeningElement ).get();
+			fixNameAttribute( openingEl.value, stateProp );
+
+			changeMethodsToAdd.add( linkPropName );
+
+			const { valuePropName, changeHandlerName } = linkPropInfo( linkPropName );
+
+			openingEl.value.attributes.push( j.jsxAttribute(
+				j.jsxIdentifier( valuePropName ),
+				j.jsxExpressionContainer( j.template.expression`this.state.${ stateProp }` )
+			) );
+
+			openingEl.value.attributes.push( j.jsxAttribute(
+				j.jsxIdentifier( 'onChange' ),
+				j.jsxExpressionContainer( j.template.expression`this.${ changeHandlerName }` )
+			) );
+
+			linkProps.remove();
+			changedSomething = true;
+		} );
+
+		return changeMethodsToAdd;
+	}
+
+	// Add change handler methods (i.e., `handleInputValueChange`) to the class
+	function addChangeMethods( classPath, changeMethodsToAdd ) {
+		const objectExpr = classPath.value.arguments[ 0 ];
+
+		for ( const linkPropName of changeMethodsToAdd ) {
+			const { valuePropName, changeHandlerName } = linkPropInfo( linkPropName );
+			const property = j.property(
+				'init',
+				j.identifier( changeHandlerName ),
+				j.functionExpression(
+					null,
+					[ j.identifier( 'e' ) ],
+					j.blockStatement( [
+						j.template.statement`this.setState( { [ e.currentTarget.dataset.name ]: e.currentTarget.${ valuePropName } } )`
```

Comment:
How does this work? [`dataset`](https://developer.mozilla.org/de/docs/Web/API/HTMLElement/dataset) requires the presence of a `data-` attribute, no?